From: Jiri Slaby <jslaby@suse.cz>
Subject: sysfs: spectre_v2, handle spec_ctrl
Patch-mainline: never, SUSE specific
References: bsc#1075994 bsc#1075091

/sys/devices/system/cpu/vulnerabilities/spectre_v2 does no take into
account that we have IBRS+IBPB in our tree. Fix that.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 arch/x86/kernel/cpu/bugs.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/x86/kernel/cpu/bugs.c
+++ b/arch/x86/kernel/cpu/bugs.c
@@ -330,6 +330,8 @@ ssize_t cpu_show_spectre_v2(struct devic
 {
 	if (!boot_cpu_has_bug(X86_BUG_SPECTRE_V2))
 		return sprintf(buf, "Not affected\n");
+	if (boot_cpu_has(X86_FEATURE_SPEC_CTRL))
+		return sprintf(buf, "Mitigation: IBRS+IBPB\n");
 
 	return sprintf(buf, "%s%s\n", spectre_v2_strings[spectre_v2_enabled],
 		       spectre_v2_module_string());
